.github/workflows/build-test-radosgw.yml

```diff
@@ -113,3 +133,98 @@ jobs:
           ./sfs-smoke-test.sh 127.0.0.1:7480
 
           docker kill $CONTAINER
+
+  run-s3gw-tests:
+    strategy:
+      matrix:
+        op:
+          - create
+          # - list
+          - delete
+        resource:
+          - bucket
+          - object
+
+    runs-on: self-hosted
+    needs:
+      - build-and-test
+
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: ceph
+
+      - name: Checkout s3tests
+        uses: actions/checkout@v3
+        with:
+          repository: ceph/s3-tests
+          path: s3tests
+          branch: ceph-master
+
+      - name: Checkout s3gw Infrastructure
+        uses: actions/checkout@v3
+        with:
+          path: s3gw
+          repository: aquarist-labs/s3gw
+
+      - name: Download Radosgw Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ needs.build-and-test.outputs.artifact_id }}
+
+      - name: Unpack Artitfacts
+        run: |
+          tar -xvf build-results.tar.gz
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+
+      - name: Install dependencies
+        run: |
+          YQ_GH_URL=https://github.com/mikefarah/yq
+          sudo apt-get update
+          sudo apt-get install -y \
+            python3-nose \
+            wget
+
+          # Unfortunately, since yq is only available through snap on Ubunut and
+          # that doesn't work in docker containers (at least not out of the
+          # box), this abomination is the way to go to install yq.
+          sudo wget \
+            "${YQ_GH_URL}/releases/latest/download/yq_linux_amd64" \
+            -O /usr/bin/yq && \
+            sudo chmod +x /usr/bin/yq
+
+          python3 -m pip \
+            install --upgrade -r s3tests/requirements.txt
+
+      - name: Run S3tests ${{ matrix.op }} ${{ matrix.resource }}
+        run: |
+          set -x
+
+          export CEPH_DIR=${GITHUB_WORKSPACE}/ceph
+          export OUTPUT_DIR=${GITHUB_WORKSPACE}/s3test-results
+          export S3TEST_REPO=${GITHUB_WORKSPACE}/s3tests
+          export FORCE_CONTAINER=ON
+
+          export FIXTURES=$GITHUB_WORKSPACE/ceph/qa/rgw/store/sfs/tests/fixtures
+          export S3TEST_CONF=$FIXTURES/s3tests.conf
+
+          pushd s3tests
+          for t in $(yq \
```

Comment:
the yq dependency seems more trouble than its worth, especially with the rather simple 'get map key, output as space sep list' usage.

Consider python instead:

```
python <<EOF
from yaml import load
print(" ".join(load(open('s3-tests.yaml'))["bucket_create"]))
EOF

```

use either with for t in ..
or better use `\n` and pipe to while read t; do ..

Reply:
this also presumes 'PyYaml' as an installed dependency though. I agree with not installing "random" files from the internet, but doing it with Python will require the dependency to be installed.

Reply:
`yq` is needed for the s3tests-runner script anyways, as it is what I use to compile an aggregated report about which tests pass and which fail. I'm not keen on re-implementing what I need from it in python, especially not the more complex parts like inserting into maps and converting to JSON. That being said, I pinned the version down with a checksum check for good measure.

Reply:
:+1: